app/integration/runtime/src/test/java/io/syndesis/integration/runtime/IntegrationTestSupport.java

```diff
@@ -19,21 +19,37 @@
 import java.util.Arrays;
 import java.util.Collections;
 import javax.xml.bind.JAXBException;
-
-import io.syndesis.common.model.integration.Flow;
-import io.syndesis.common.model.integration.Integration;
-import io.syndesis.common.model.integration.Step;
 import org.apache.camel.CamelContext;
 import org.apache.camel.model.ModelHelper;
 import org.apache.camel.model.ProcessorDefinition;
 import org.apache.camel.model.RouteDefinition;
 import org.apache.camel.model.RoutesDefinition;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.syndesis.common.model.integration.Flow;
+import io.syndesis.common.model.integration.Integration;
+import io.syndesis.common.model.integration.Step;
+import io.syndesis.common.util.StringConstants;
 
-public class IntegrationTestSupport {
+public class IntegrationTestSupport implements StringConstants {
     private static final Logger LOGGER = LoggerFactory.getLogger(IntegrationTestSupport.class);
 
+    protected ObjectMapper mapper = new ObjectMapper();
+
+    protected static class MsgSnippet {
```

Comment:
Not very sure to understand the need of this msg snippet thing, isn't a map and/or Jackson enough ?

Reply:
A MsgSnippet is simply a key/value pair object rather than a map so more akin to a Map.Entry. In the tests it allows to generate snippets and adds snippets as the value of snippets (so recursively). Basically, just reducing complexity and verbosity of test setup (started with maps and it was getting rather cluttered).

Reply:
Os so I'd try to find a better name as `message` has a very well defined meaning in Camel and it may confuse if used with different meaning (same for the utility methods like `jsonMessage(MsgSnippet...)` and `message(MsgSnippet...)`

Reply:
ok... renamed.